src/lib/fragments/AgGrid.react.js

```diff
@@ -413,6 +413,9 @@ export default class DashAgGrid extends Component {
                         ? this.suppressGetDetail(value.detailColName)
                         : this.callbackGetDetail;
                 }
+                if ('detailGridOptions' in value) {
+                    value.detailGridOptions.components = this.state.components;
```

Comment:
We shouldn't be mutating `value` (here or in the block above `value.getDetailRowData = ...`, sorry I hadn't noticed this before) - so we should do something like:
```js
let adjustedVal = value;
if ('suppressCallback' in value) {
    adjustedVal = {
        ...adjustedval,
        getDetailRowData: value.suppressCallback
                        ? this.suppressGetDetail(value.detailColName)
                        : this.callbackGetDetail
    };
}
if ('detailGridOptions' in value) {
    adjustedVal = {
        ...adjustedVal,
        detailGridOptions: {
            ...value.detailGridOptions,
            components: this.state.components
        }
    };
}
```

Reply:
Can also do this with https://ramdajs.com/docs/#assoc and `assocPath` if you prefer, might be more compact.